test/test_tokenization.py

```diff
@@ -272,8 +271,8 @@ def test_fast_bert_tokenizer_strip_accents(caplog):
                                use_fast=True,
                                strip_accents=False)
     assert type(tokenizer) is BertTokenizerFast
+    assert tokenizer.do_lower_case
     assert tokenizer._tokenizer._parameters['strip_accents'] is False
```

Comment:
Hey @PhilipMay I am confused about the usage of `strip_accents` and how it is used in the fast tokenizers. Since you added it to transformers could you have a look at this failing test?

Reply:
Sure. :-) 

This is the problem right?
```
caplog = <_pytest.logging.LogCaptureFixture object at 0x7fa228117a50>

    def test_fast_bert_tokenizer_strip_accents(caplog):
        caplog.set_level(logging.CRITICAL)
    
        tokenizer = Tokenizer.load("dbmdz/bert-base-german-uncased",
                                   use_fast=True,
                                   strip_accents=False)
        assert type(tokenizer) is BertTokenizerFast
        assert tokenizer.do_lower_case
>       assert tokenizer._tokenizer._parameters['strip_accents'] is False
E       AttributeError: 'tokenizers.Tokenizer' object has no attribute '_parameters'
```

The old tokenizer implementation was like this:

- if ippercase then do not strip accents
- if lowercase then always strip accents

I wanted to bo able to do lower case but not to strip accents. Do I introduced a new param called `strip_accents` - see #6280

`strip_accents` has the default value of `None` which results in the "old" behavior. This is because I did not want to introduce a breaking change.

See here:
```
            **strip_accents**: (`optional`) boolean (default None)
                Whether to strip all accents. If this option is not specified (ie == None),
                then it will be determined by the value for `lowercase` (as in the original Bert).
```

If `strip_accents` is set to True or False it works like specified by the default value.
Because `dbmdz/bert-base-german-uncased` is an old model and does not specify `strip_accents` in its tokenizer config file thing the value should be at default (`None`).

So you should change it it `assert tokenizer._tokenizer._parameters['strip_accents'] is None`

And introduce a 2nd test where you use our electra model `german-nlp-group/electra-base-german-uncased` and test for `assert tokenizer._tokenizer._parameters['strip_accents'] is False`.

Reply:
Ah ok, I understand the parameter better now. thanks.

But the error is coming from `'tokenizers.Tokenizer' object has no attribute '_parameters'`. It seems to me as something has changed fundamentally in the tokenizer classes. Do you agree?

Reply:
I do not think so. The FastTokenizer is just implemented in RUST and you can not access the internal properties with Python code.

AFAIK it is set here: https://github.com/huggingface/transformers/blob/master/src/transformers/tokenization_bert_fast.py#L199

Reply:
I do not think so. The FastTokenizer is just implemented in RUST and you can not access the internal properties with Python code.

AFAIK it is set here: https://github.com/huggingface/transformers/blob/master/src/transformers/tokenization_bert_fast.py#L199

But `type(pre_tok_class)` is `tokenizers.normalizers.BertNormalizer` which is not implemented with Python.
See here: https://github.com/huggingface/tokenizers/blob/master/bindings/python/py_src/tokenizers/normalizers/__init__.py#L4

And here: https://github.com/huggingface/tokenizers/blob/fff856cff76c1cb86d3f1ec392b5313f93b76ce0/bindings/python/py_src/tokenizers/normalizers/__init__.pyi#L18

Reply:
Ok, does this mean I can delete the test?

To confirm, the test was passing before on the `BertTokenizerFast` and somehow accessed the "_parameters" attribute from within python. So you are saying that a lot of the parts in the BertTokenizerFast that were previously in python now got ported to the rust part and are not accessible?

Reply:
That test was green with transformers version 3.4.0 right? And you are 100% sure that it did load the fast tokenizer? Or could it be that the slow tokenizer somehow sneaked in?

Reply:
Exactly, that test is green right now and there is an assertion that the fast tokenizer is used, so I guess it loads it as well.